api/v1/env.go

```diff
@@ -1,9 +1,8 @@
 package v1
 
-// +kubebuilder:validation:XValidation:message="name must match [a-zA-Z_][a-zA-Z0-9_]*",rule="self.name.matches('^[a-zA-Z_][a-zA-Z0-9_]*$')"
```

Comment:
CEL budgets are very annoying, and I don't want to limit the size of the variations array. So I'll delete this validation for now. We can consider using the eno-controller as a validating webhook in the future.